File: src/js/postgres_block.js
const { gaussianCDF } = require("./gaussian"),
  { Pool } = require("pg");

void (async function () {
  let query = "",
    pool,
    conn;
  try {
    pool = new Pool({
      host: "localhost",
      database: "gaussian",
      user: "root",
      password: "root123",
    });
    conn = await pool.connect();

    conn.query(
      "DROP TABLE IF EXISTS js_block; CREATE TABLE js_block(id SERIAL " +
        "PRIMARY KEY, z_score REAL NOT NULL, cumulative_distribution REAL NOT NULL);"
    );

    for (let i = -500; i <= 500; i++) {
      let x = i / 100,
        prob = gaussianCDF(0, 1, x);
      query += `INSERT INTO js_block(z_score, cumulative_distribution) VALUES (${x}, ${prob.toFixed(
        6
      )});`;
    }
    conn.query(query, () => conn.end());
  } catch (err) {
    console.log(err.message);
  }
})();

File: src/py/postgres_block.py
from gaussian import gaussianCDF
from psycopg2 import connect

query = ''
try:
	conn = connect(
		host="localhost",
		user="root",
		password="root123",
		database="gaussian",
	)
	conn.autocommit = True
	cur = conn.cursor()
	cur.execute('''DROP TABLE IF EXISTS tb_py_block; CREATE TABLE tb_py_block(id SERIAL PRIMARY KEY,
	z_score REAL NOT NULL, cumulative_distribution REAL NOT NULL);''')
	for i in range(-500, 501):
		x = i / 100
		prob = gaussianCDF(0, 1, x)
		query += f'INSERT INTO tb_py_block(z_score, cumulative_distribution) VALUES ({x:.2f}, {prob:.6f});'
	cur.execute(query)
	conn.close()
except Exception as e:
	print(e)
